utils/results/results.go

```diff
@@ -0,0 +1,315 @@
+	// General fields describing the command metadata
+	XrayVersion    string `json:"xray_version"`
+	EntitledForJas bool   `json:"jas_entitled"`
+	// MultiScanId is a unique identifier that is used to group multiple scans together.
+	MultiScanId string `json:"multi_scan_id,omitempty"`
+	// Results for each target in the command
+	Targets      []*TargetResults `json:"targets"`
+	targetsMutex sync.Mutex       `json:"-"`
+	// Error that occurred during the command execution
+	Error error `json:"error,omitempty"`
+}
+
+type ScanTarget struct {
+	// Physical location of the target: Working directory (audit) / binary to scan (scan / docker scan)
+	Target string `json:"target,omitempty"`
+	// Logical name of the target (build name / module name / docker image name...)
+	Name string `json:"name,omitempty"`
+	// Optional field (not used only in build scan) to provide the technology of the target
+	Technology techutils.Technology `json:"technology,omitempty"`
+}
+
+type TargetResults struct {
+	ScanTarget
+	// All scan results for the target
+	ScaResults *ScaScanResults  `json:"sca_scans,omitempty"`
+	JasResults *JasScansResults `json:"jas_scans,omitempty"`
+	// Errors that occurred during the scans
+	Errors      []error    `json:"errors,omitempty"`
+	errorsMutex sync.Mutex `json:"-"`
+}
+
+type ScaScanResults struct {
+	IsMultipleRootProject *bool `json:"is_multiple_root_project,omitempty"`
+	// Target of the scan
+	Descriptors []string `json:"descriptors,omitempty"`
+	// Sca scan results
+	XrayResults []services.ScanResponse `json:"xray_scan"`
+}
+
+type JasScansResults struct {
+	ApplicabilityScanResults []*sarif.Run `json:"contextual_analysis,omitempty"`
+	SecretsScanResults       []*sarif.Run `json:"secrets,omitempty"`
+	IacScanResults           []*sarif.Run `json:"iac,omitempty"`
+	SastScanResults          []*sarif.Run `json:"sast,omitempty"`
+}
+
+func NewCommandResults(xrayVersion string, entitledForJas bool) *SecurityCommandResults {
+	return &SecurityCommandResults{XrayVersion: xrayVersion, EntitledForJas: entitledForJas, targetsMutex: sync.Mutex{}}
+}
+
+func (r *SecurityCommandResults) SetMultiScanId(multiScanId string) *SecurityCommandResults {
+	r.MultiScanId = multiScanId
+	return r
+}
+
+// --- Aggregated results for all targets ---
+
+func (r *SecurityCommandResults) GetTargetsPaths() (paths []string) {
+	for _, scan := range r.Targets {
+		paths = append(paths, scan.Target)
+	}
+	return
+}
+
+func (r *SecurityCommandResults) GetScaScansXrayResults() (results []services.ScanResponse) {
+	for _, scan := range r.Targets {
+		results = append(results, scan.GetScaScansXrayResults()...)
+	}
+	return
+}
+
+func (r *SecurityCommandResults) GetJasScansResults(scanType jasutils.JasScanType) (results []*sarif.Run) {
+	if !r.EntitledForJas {
+		return
+	}
+	for _, scan := range r.Targets {
+		results = append(results, scan.GetJasScansResults(scanType)...)
+	}
+	return
+}
+
+func (r *SecurityCommandResults) GetErrors() (err error) {
```

Comment:
This will create a single error for all targets. dont we want to split them somehow? even in the text itself...
Maybe we do something like this:
"Target <SecurityCommandResults.Targets[i].ScanTarget.Name> errors: %s \n"
In this way we can have more cleare error log in case we have several targets